File: modules/purplejs-core/src/test/java/io/purplejs/core/script/ResolveTest.java
package io.purplejs.core.script;

import org.junit.Test;

import io.purplejs.core.resource.ResourcePath;
import io.purplejs.core.value.ScriptExports;

import static org.junit.Assert.*;

public class ResolveTest
    extends AbstractScriptTest
{
    @Test
    public void resolve()
    {
        final ResourcePath path = ResourcePath.from( "/app/resolve/resolve-test.js" );
        final ScriptExports exports = run( path );

        assertNotNull( exports );
        assertEquals( path, exports.getResource() );
    }
}

File: modules/purplejs-core/src/test/resources/app/resolve/resolve-test.js
/* global Java */
/* global resolve */
var assert = Java.type('org.junit.Assert');

require('./other/resolve-in-require').test();
require('./other/resolve-in-require').test();

var relative = resolve('./test.html').toString();
assert.assertEquals('/app/resolve/test.html', relative);

var absolute = resolve('/app/test.html').toString();
assert.assertEquals('/app/test.html', absolute);
